minibear/src/main/java/frc/robot/commands/IntakeAutoCommand.java

```diff
@@ -0,0 +1,110 @@
+// ************************************************************
+// Bishop Blanchet Robotics
+// Home of the Cybears
+// FRC - Charged Up - 2023
+// File: EveryBotPickerAutoUptakeCommand.java
+// Intent: Forms a command to automatically uptake the cargo into the every bot picker.
+// ************************************************************
+
+// ʕ •ᴥ•ʔ ʕ•ᴥ•  ʔ ʕ  •ᴥ•ʔ ʕ •`ᴥ´•ʔ ʕ° •° ʔ ʕ •ᴥ•ʔ ʕ•ᴥ•  ʔ ʕ  •ᴥ•ʔ ʕ •`ᴥ´•ʔ ʕ° •° ʔ 
+
+
+package frc.robot.commands;
+
+import edu.wpi.first.wpilibj2.command.CommandBase;
+import frc.robot.common.WristPosition;
+import frc.robot.subsystems.IntakeSubsystem;
+import frc.robot.subsystems.WristSubsystem;
+import edu.wpi.first.wpilibj.Timer;
+
+
+public class IntakeAutoCommand extends CommandBase {
+
+    private IntakeSubsystem intakeSubsystem;
+    private WristSubsystem wristSubsystem;
+    
+    private Timer timer = new Timer();
+    private boolean done = false;
+    private final double moveToPositionTimeStamp = 1.7; //TODO set times
+    private final double shootTimeStamp = 3;
+    private final double shootDoneTimeStamp = 3.5;
+
+    private double relativeSpeed;
+    private double durationSeconds;
+
+    /**
+     * Constructor for EveryBotPickerAutoUptakeCommand
+     * @param intakeSubsystem - the subsystem for the everybot picker
+     * @return 
+     */
+    public IntakeAutoCommand(IntakeSubsystem intakeSubsystem, WristSubsystem wristSubsystem) {
+        this.intakeSubsystem = intakeSubsystem;
+        addRequirements(this.intakeSubsystem);
+
+        this.wristSubsystem = wristSubsystem;
+        addRequirements(this.wristSubsystem);
+    }
+
+    /**
+     * Override of the initialize method on CommandBase - called on start of command
+     */
+    @Override
+    public void initialize() {
+        timer.reset();
+        timer.start();
+        done = false;
+    }
+
+    /**
+     * Override of the execute method on CommandBase - Called every time the scheduler runs while the command is scheduled.
+     */
+    @Override
+    public void execute() {
+        if (timer.hasElapsed(moveToPositionTimeStamp)) {
+            new WristPositionCommand(wristSubsystem, WristPosition.PositionTwo);
+        } 
+
+        if (timer.hasElapsed(shootTimeStamp)) {
+            new IntakeDefaultCommand(wristSubsystem,
+            intakeSubsystem, 
+                    () -> 0.0, // No uptake
+                    () -> 1.0); // Assuming full power expelling
+        }
+
+        if (timer.hasElapsed(shootDoneTimeStamp)) {
+            new IntakeDefaultCommand(wristSubsystem,
+            intakeSubsystem, 
+                    () -> 0.0, // No uptake
+                    () -> 0.0); // No uptake
+
+        new WristPositionCommand(wristSubsystem, WristPosition.PositionThree);
+
+        }
+
+        done = true;
```

Comment:
Owen - you need to protect this and various other things above (like each of the commands) from changing until the execution of each is done.  The method 'execute()' is a non-blocking call, that means that it is supposed to be called every 20 milliseconds and while each of the actions is progressing, the execute() method needs to keep some amount of state around how each of the actions it is doing is progressing.  Only when each of the actions is complete should it set 'done = true'.  Does that make sense?  Do you want help here on how to protect each of the items here?

Reply:
Oh, that makes sense. Do I need checks for each or can I just get rid of that done = true; line? Its fine if its just in the isFinished() method right?